### PR TITLE
Moved all prox and solver source code

### PR DESCRIPTION
**Pull request by stephanegaiffas**
Second iteration for the reorganization and flattening of tick, see #124 
- Moved all `prox` and `solver` python modules and C++ source code
- Updated config files for compilation with mkn
- Updated and reorganized documentation

### CODE REVIEW

File: tick/optim/prox/build/__init__.py [hunk 1]
```diff
@@ -1 +0,0 @@
-# License: BSD 3 clause
```
Comment (Mbompr) [inline tick/optim/prox/build/__init__.py]: This file must be kept for pip distribution (and moved to ` tick/prox/build/__init__.py`)

File: tick/optim/solver/build/__init__.py [hunk 2]
```diff
@@ -1 +0,0 @@
-# License: BSD 3 clause
```
Comment (Mbompr) [inline tick/optim/solver/build/__init__.py]: This file must be kept for pip distribution (and moved to ` tick/solver/build/__init__.py`)

**Comment by Mbompr** [comment]
Without these files I get 
`ImportError: No module named 'tick.prox.build.prox'` error as well.

I think you have these files on your computer but you need to tell git to track them manually as they are ignored. Either by typing `git add -f tick/prox/build/__init__.py` either by tweaking .gitignore file, this would be cleaner I don't know how to set a general rule for all `__init__.py` under `build` folders.

**Comment by Dekken** [comment]
Is this superseded  by #143 ?

**Comment by stephanegaiffas** [comment]
This is superseded by #145

**Pull request closed without merging**